spec/models/orchestration_template_spec.rb

```diff
@@ -36,6 +36,24 @@
         expect(@existing_record).not_to eq(OrchestrationTemplate.find_or_create_by_contents(@query_hash)[0])
         expect(OrchestrationTemplate.count).to eq(2)
       end
+
+      it "uses subclass if type is present" do
+        expect(ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate).to receive(:calc_md5).at_least(:once)
+        expect(described_class).not_to receive(:calc_md5)
+
+        @query_hash[:draft] = false
+        @query_hash[:type] = ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate.name
+        OrchestrationTemplate.find_or_create_by_contents(@query_hash)
+      end
+
+      it "uses parent if type is not present" do
+        expect(ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate).not_to receive(:calc_md5)
+        expect(described_class).not_to receive(:calc_md5).at_least(:once)
```

Comment:
why `not_to`? shouldn't it call the parent to calculate the md5?

Reply:
Ha! Typo! But it seems like `.at_least(:once)` makes it ignore the `not_to` since it works both with

```
.not_to receive(:calc_md5).at_least(:once)
```

or with

```
.to receive(:calc_md5).at_least(:once)
```

Thanks for noticing, I've updated the code.